events/sse_events.go

```diff
@@ -0,0 +1,144 @@
+package events
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+)
+
+// Event holds state of parsed fields from marathon EventStream
+type SSEEvent struct {
+	Type        string
+	Timestamp   Timestamp
+	Body        []byte
+	ID          string
+	Delay       string
+	maxLineSize int64
+}
+
+func (e *SSEEvent) String() string {
+	return fmt.Sprintf("Type: %s, Body: %s", e.Type, string(e.Body))
+}
+
+func (e *SSEEvent) parseLine(line []byte) bool {
+	// https://www.w3.org/TR/2011/WD-eventsource-20110208/
+	// Quote: Lines must be separated by either a U+000D CARRIAGE RETURN U+000A
+	// LINE FEED (CRLF) character pair, a single U+000A LINE FEED (LF) character,
+	// or a single U+000D CARRIAGE RETURN (CR) character.
+
+	//If the line is empty (a blank line)
+	if len(line) == 0 || bytes.Compare(line, []byte("\n")) == 0 {
+		//Dispatch the event, as defined below.
+		return !e.isEmpty()
+	}
+
+	//If the line starts with a U+003A COLON character (:)
+	if bytes.HasPrefix(line, []byte{':'}) {
```

Comment:
Good catch, I wrote benchmark for this. Average times spent parsing single data line was ~10ns better. So it makes worth it :).
I have made appropriate change to the code.